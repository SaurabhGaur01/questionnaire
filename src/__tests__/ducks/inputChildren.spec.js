import {
    setChildren,
    clearChildren,
    inputChildrenReducer,
} from '../../ducks/userInput/inputChildren';

describe('inputChildren Duck Tests', () => {
    const mockObject = 1;

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setChildren should match the snapshot' , () => {
            expect(setChildren(mockObject)).toMatchSnapshot();
        });
        it('clearChildren should match the snapshot' , () => {
            expect(clearChildren()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(inputChildrenReducer()).toEqual(0);
        });
        it('should return the passed state if called with no action' , () => {
            expect(inputChildrenReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(inputChildrenReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(inputChildrenReducer('', setChildren(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearChildren' , () => {
            expect(inputChildrenReducer('', clearChildren())).toEqual(0);
        });
    });
});