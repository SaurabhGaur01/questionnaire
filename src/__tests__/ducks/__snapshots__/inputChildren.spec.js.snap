// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputChildren Duck Tests Action Generator clearChildren should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_CHILDREN",
}
`;

exports[`inputChildren Duck Tests Action Generator setChildren should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_CHILDREN",
  "value": 1,
}
`;
